File: src/main/java/atlix/logic/services/SupplierService.java
package atlix.logic.services;

import atlix.model.beans.Supplier;
import atlix.model.repositories.SupplierRepository;
import atlix.model.response.SupplierDTO;
import jakarta.transaction.Transactional;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;

@Service
public class SupplierService {

    private final SupplierRepository supplierRepository;

    public SupplierService(SupplierRepository supplierRepository) {
        this.supplierRepository = supplierRepository;
    }

    public List<SupplierDTO> findAll() {
        return supplierRepository.findAll()
                .stream()
                .map(SupplierDTO::toDTO)
                .toList();
    }

    @Transactional
    public SupplierDTO findByName(String name) {
        return supplierRepository.findByName(name)
                .map(SupplierDTO::toDTO)
                .orElseThrow(() -> new IllegalArgumentException("Supplier with name " + name + " not found"));
    }

    @Transactional
    public void save(SupplierDTO supplierDTO) {
        var p = supplierRepository.findByName(supplierDTO.name());

        if (p.isPresent()) {
            throw new IllegalArgumentException("Supplier with name " + supplierDTO.name() + " already exists");
        }

        var  supplier = new Supplier();
        supplier.setName(supplierDTO.name());
        supplier.setEmail(supplierDTO.email());
        supplier.setAddress(supplierDTO.address());
        supplier.setPhone(supplierDTO.phone());
        supplier.setProducts(new ArrayList<>());

        supplierRepository.save(supplier);
    }

    @Transactional
    public void delete(SupplierDTO supplierDTO) {
        var p  = supplierRepository.findByName(supplierDTO.name());
        if (p.isEmpty()) {
            throw new IllegalArgumentException("Supplier with name " + supplierDTO.name() + " not found");
        }

        supplierRepository.delete(p.get());
    }
}

File: src/main/kotlin/atlix/logic/services/SupplierService.kt
package atlix.logic.services

import atlix.model.repository.SupplierRepository
import atlix.util.WindowLoader
import javafx.fxml.LoadException

class SupplierService {

    private val supplierRepository: SupplierRepository = SupplierRepository()
    private val windowLoader = WindowLoader()

    fun closeWindow(stage: javafx.stage.Stage) = windowLoader.closeWindow(stage)

    fun loadMainView() {
        try {
            WindowLoader().showWindow(atlix.util.Paths.MAIN_VIEW, "Vista Principal", false)
        } catch (e: LoadException) {
            e.printStackTrace()
            throw RuntimeException("Error al cargar la vista principal: ${e.message}")
        }
    }

    fun addNewSupplier(name: String, company: String, phone: String, email: String, nameContact: String) {
        val supplier = atlix.model.beans.SupplierBean(
            name = name,
            company = company,
            numberPhone = phone,
            email = email,
            nameContact = nameContact
        )
        supplierRepository.save(supplier)
    }

    fun getProductsForSupplier(supplierId: Int): List<atlix.model.content.ProductSupplier> {
        return supplierRepository.findById(supplierId)?.products?.toList() ?: emptyList()
    }
}